go/api/base_client.go

```diff
@@ -1204,3 +1206,23 @@ func (client *baseClient) Renamenx(key string, newKey string) (Result[bool], err
 	}
 	return handleBooleanResponse(result)
 }
+
+func (client *baseClient) XAdd(key string, values [][]string) (Result[string], error) {
```

Comment:
I see that there are multiple variants for `XAdd` https://github.com/valkey-io/valkey-glide/blob/daa5af2f6f1f3ebc55dce2c7e461cf4a954d6ec4/java/client/src/main/java/glide/api/BaseClient.java#L2708-L2776. Are we planning to create separate PRs for those?

Reply:
Will do in scope of this PR.
I was waiting for discussion about `zadd` options to complete to follow the choosen pattern.

Reply:
Do you want me to add `xadd` with map argument? `xadd` supports repeating field names, see such case in the test I added.
For the same reason `xread` returns arrays of k-v pairs instead of maps.

Reply:
I think the current implementation should suffice.